### PR TITLE
[#2607] Removing unnecessary Save step in workflow

### PR DESCRIPTION
**Pull request by Maurier**

### CODE REVIEW

**Comment by hydrobuild** [comment]
Build finished.

**Comment by Maurier** [comment]
@aphelionz This is ready for review now.

File: theme/static/js/generic-resource.js [hunk 1]
```diff
@@ -222,34 +227,38 @@ function showAddEditExtraMetaPopup(edit, row_id_str) {
...
+            "The name already exists. Please input a different name.</div>");
         $("#extra_meta_msg").show();
         return;
     }
 
-    if(foundDuplicatedName(t, extra_meta_name, edit_extra_meta_row_id)) {
-        $("#extra_meta_msg").html("<font color='red'>The name already exists. Please input a different name.</font>");
+    if (extra_meta_name.length == 0 || extra_meta_value.length == 0) {
+        $("#extra_meta_msg").html("<div class='alert alert-danger'>" +
+            "Both name and value are required fields that cannot be left blank.</div>");
```
Comment (aphelionz) [inline theme/static/js/generic-resource.js]: @Maurier Is there also server-side checking for this as well?

Comment (Maurier) [reply theme/static/js/generic-resource.js]: @aphelionz I disabled front end validations to test for this. I found that validation for empty required fields is not working properly. The ajax call returns an error, but the submission actually goes through.

Validations for unique constraints are working properly.

Comment (pkdash) [reply theme/static/js/generic-resource.js]: @aphelionz and @Maurier I see this PR is targeted for current RC. Wondering why this feature needs to be part of the current RC? No new features should be added to any RC.

Comment (aphelionz) [reply theme/static/js/generic-resource.js]: Discussed this on the call, please leave in the RC

**Comment by Maurier** [comment]
@aphelionz the back end validations have been put in place now as well. 

While working on this I found out that some error alerts, shown as product of failed validations, in the front end were showing on green background, which indicates success. Since this was misleading I went ahead and improved our scripts to handle custom alerts appropriately.

We can now show custom alerts of the following types: success, error, info.  A title can also be specified for them.

**Comment by hydrobuild** [comment]
Build finished.

**Comment by hydrobuild** [comment]
Build finished.

**Comment by hydrobuild** [comment]
Build finished.

**Pull request merged**